src/pages/case-studies/index.tsx

```diff
@@ -0,0 +1,131 @@
+  const { t } = useTranslation();
+
+  // Sort by date
+  caseStudiesMdx.sort((a: any, b: any) => {
+    return Date.parse(b.frontmatter.dates[0]) - Date.parse(a.frontmatter.dates[0]);
+  });
+
+  return (
+    <Layout title={t('case-studies:seo.title')} desc={t('case-studies:seo.desc')}>
+      <Container layout="wide" dark={true} paddingY={false}>
+        <Container gridCols="12" gridGutters="11" flex={true} paddingX={false}>
+          <Hero
+            title={t('case-studies:hero.title')}
+            subtitle={t('case-studies:hero.subtitle')}
+          ></Hero>
+        </Container>
+      </Container>
+
+      <Container paddingY={false}>
+        <Container gridCols="10" gridGutters="9">
+          <TextContent>
+            {caseStudiesMdx.map(({ url, frontmatter }, i: number) => {
+              const title = (
+                <h4 className="mb__0">
+                  <Link href={url}>{frontmatter.title}</Link>
+                </h4>
+              );
+              const subtitle = (
+                <div className="text__smallcaps mt__2XS">
+                  {getCaseStudySubtitle(
+                    frontmatter.service,
+                    frontmatter.terms_types,
+                    frontmatter.dates
+                  )}
+                </div>
+              );
+              const separator = caseStudiesMdx.length - 1 === i ? '' : <hr className="mt__XL" />;
+
+              return (
+                <div key={frontmatter.title} className="mb__XL">
+                  {title}
+                  {subtitle}
+                  {separator}
+                </div>
+              );
+            })}
+          </TextContent>
+        </Container>
+      </Container>
+    </Layout>
+  );
+}
+
+export const getStaticProps: GetStaticProps = async (props) => {
+  const locale = props.locale || 'en';
+  const { files: caseStudiesFilesPath } = getI18nContentFilePaths('pages', locale, {
+    subfolder: 'case-studies',
+    extension: false,
+  });
+
+  const caseStudiesMdx: CaseStudy[] = [];
+
+  await Promise.all(
+    caseStudiesFilesPath.map(async (filename) => {
+      const mdxContent = await loadMdxFile(
+        {
+          filename,
+          folder: `pages`,
+        },
+        locale
+      );
+      caseStudiesMdx.push({ url: `/${filename}`, frontmatter: mdxContent?.frontmatter || {} });
+    })
+  );
+
+  return {
+    props: {
+      caseStudiesMdx,
+      ...props,
+    },
+  };
+};
+
+export function getCaseStudySubtitle(service: string, termsTypes: [], dates: []) {
+  const router = useRouter();
+
+  //Map dates by years-months
+  let orderedDates = new Map();
+  dates.forEach((date: string) => {
+    const yearsMonths = dayjs(date).format('YYYY-MM');
+    const day = dayjs(date).format('DD');
+    let yearMonthsGroup = orderedDates.get(yearsMonths) ?? [];
+    yearMonthsGroup.push(day);
+    orderedDates.set(yearsMonths, yearMonthsGroup);
+  });
```

Comment:
If I understand correctly, you want to make sure that multiple dates are ordered chronologically.
I'm honestly not sure it is worth investing that much in a use case that is very rare (multiple dates in a case study).
If you want to do this though, might I suggest relying on native date comparisons and Array functions instead? 🙂

```suggestion
  dates.sort((date1, date2) => date1 - date2});
```

Reply:
This is not exactly what I am trying to do here.

From an array of dates `[ '2021-11-03', '2021-11-10', '2021-11-16' ]` i'm build a Map of `year-month` like `Map(1) { '2021-11' => [ '03', '10', '16' ] }`

This allows me to group the display of dates belonging to the same year/month into a single string.

Instead of  `03 novembre 2021, 10 novembre 2021, 16 novembre 2021` i can display `03,10,16 novembre 2021`

I totally agree that it's a special case, but I couldn't ignore it (my eyes hurt too much). Now that it's done I'd rather leave it.